assembly/src/parsers/nodes/advice.rs

```diff
@@ -0,0 +1,60 @@
+use super::super::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
+use core::fmt;
+
+// ADVICE INJECTORS
+// ================================================================================================
+
+/// Instructions which inject data into the advice provider.
+///
+/// These instructions can be used to perform two broad sets of operations:
+/// - Push new data onto the advice stack.
+/// - Insert new data into the advice map.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum AdviceInjector {
+    PushU64div,
+    PushMapVal,
+    PushExt2inv,
+    PushExt2intt,
+    PushSmtGet,
+    InsertMem,
+}
+
+impl Serializable for AdviceInjector {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        match self {
+            AdviceInjector::PushU64div => target.write_u8(0),
+            AdviceInjector::PushMapVal => target.write_u8(1),
+            AdviceInjector::PushExt2inv => target.write_u8(2),
+            AdviceInjector::PushExt2intt => target.write_u8(3),
+            AdviceInjector::PushSmtGet => target.write_u8(4),
+            AdviceInjector::InsertMem => target.write_u8(5),
+        }
+    }
+}
+
+impl Deserializable for AdviceInjector {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        match source.read_u8()? {
+            0 => Ok(AdviceInjector::PushU64div),
+            1 => Ok(AdviceInjector::PushMapVal),
+            2 => Ok(AdviceInjector::PushExt2inv),
+            3 => Ok(AdviceInjector::PushExt2intt),
+            4 => Ok(AdviceInjector::PushSmtGet),
+            5 => Ok(AdviceInjector::InsertMem),
+            val => Err(DeserializationError::InvalidValue(val.to_string())),
```

Comment:
Yes - this could have been done similar to what you've suggested. I've kept it as is for now as some of the variants will be more complex in the future, and then we won't be able to use this simple numbering approach. I will be refactoring this specific section in #933.